packages/core/src/com/communication.ts

```diff
@@ -439,6 +446,7 @@ export class Communication {
     public envReady(instanceId: string): Promise<void> {
         const { promise, resolve } = deferred();
         if (this.readyEnvs.has(instanceId)) {
+            // the only way an environment can be in readyEnvs is if handleReady was called (why do we need to call it again?)
```

Comment:
we should answer this question

Reply:
we should. but not in this PR. I'm really not sure what's relaying on this behavior.